src/jit/lclvars.cpp

```diff
@@ -6533,9 +6533,15 @@ void Compiler::lvaDumpRegLocation(unsigned lclNum)
 #ifdef _TARGET_ARM_
     else if (varDsc->TypeGet() == TYP_DOUBLE)
     {
+#ifdef LEGACY_BACKEND
+        // The assigned registers are `lvRegNum:lvOtherReg`
         printf("%3s:%-3s    ", getRegName(varDsc->lvRegNum), getRegName(varDsc->lvOtherReg));
```

Comment:
This code looks strange to me: I think lvOtherReg should only be used if isRegPairType() is true, which is on 32-bit `LEGACY_BACKEND` targets, and then only for LONG. I'm not sure why TYP_DOUBLE would ever need to display two registers, one lvRegNum and the other lvOtherReg. Can you explain this? In the code you added below to display lvRegNum and REG_NEXT(lvRegNum), why do you need to display both for TYP_DOUBLE? Is this to display "f4:f5" for a double, instead of "d2"? Ideally, we would use the correct register numbers (sN or dN) pervasively, instead of our internal fN numbers. Until we do that, however, it seems like we should just output "fN" and "know" what it means for doubles.